### DISCUSSION TITLE
Dreame L10 Pro - High Load and OOM Killer

### DISCUSSION THREAD

**Discussion by morremeyer** [Hypfer/Valetudo#2092] [Q&A / Support] [answered]
### Are you already using Valetudo?

Yes, for a long time

### Your questions

Hey everyone,

I'm trying to find out if an issue I have is an issue with Valetudo or my specific robot and/or if anybody has any idea how I can continue debugging this issue.

The following happened/is happening:

A few days ago, our Dreame L10 Pro was not doing anything when sending commands via MQTT and the Valetudo web interface was not reachable. Since I was not at home, I asked my partner to restart it, but forgot to tell them to take it off the base.

They pressed the power button for a longer time, which didn't change anything back then - the robot seemed to neither reboot nor shut down, at least it didn't make any sounds. 

I'm not sure if that has anything to do with the issue I'm now facing, but it seems like important context.

When I returned a few days later, I tried to look into the issue and found that valetudo was not running. To get it running as it should, I took the robot off the base and shut it down. When trying to reboot, it only flashed all three LEDs constantly. 

About 1 hour later, it seemed to have shut off (it was not accessible via SSH during that time) and I tried again. This time, it seemed to boot normally.

However, every time I try to start a cleaning task now, either via MQTT or the Valetudo frontend, the following happens:

1. For about 5-10 seconds, nothing seems to happen
2. Then, the load goes up to > 20 (observed with top)
3. Nothing seems to happen again for ~20-30 seconds
4. valetudo gets killed by the OOM killer (see log output below
5. The robot starts the cleaning task
6. The load goes down again slowly

I already tried the following:

1. Creating a new firmware with dustbuilder in case this could be the issue
2. Upgrading from 2024.02.0 to 2024.06.2 in case this could be related to the MQTT fixes mentioned. Both valetudo versions behave the same in regards to my issue, though.

I'm not sure where to continue debugging this and happy about any comments.

I have attached a trace-level valetudo.log and the dmesg file from one attempt. 0 seconds dmesg is about 2024-07-28T12:07:00Z.

<details>
<summary>dmesg log lines for OOM killer</summary>

```
[  165.124463] Out of memory: Kill process 1327 (valetudo) score 891 or sacrifice child
[  165.124534] Killed process 1327 (valetudo) total-vm:416820kB, anon-rss:53044kB, file-rss:0kB, shmem-rss:0kB
[  165.162230] oom_reaper: reaped process 1327 (valetudo), now anon-rss:0kB, file-rss:0kB, shmem-rss:0kB
```

</details>

[valetudo.log](https://github.com/user-attachments/files/16403241/valetudo.log)
[dmesg.txt](https://github.com/user-attachments/files/16403242/dmesg.txt)

**Comment by Hypfer** [comment, marked as answer]
What's _probably_ happening here is that your L10 Pro is now also exhibiting the issue that is delaminating RAM chips on the PCB.
This leads to all kind of super weird memory-related behavior and unfortunately is a known issue that affects a lot of L10/Z10

If you can, RMA it for warranty replacement. If not, you either need a new PCB or reflow skills for the BGA RAM

**Reply by morremeyer** [reply]
Thank you!

I’ll contact Dreame and will see where it goes.
